### ISSUE TITLE
Hidden field value set from cookie not being URL-decoded

### ISSUE THREAD

**Issue by amphibian** [verbb/formie#1571]
### Describe the bug

When setting a hidden input's value from a cookie (which is be default URL-encoded), the value is not being URL-decoded before being inserted as the field's `value`, leading to any values containing spaces being displayed with our good friend `%20`.

### Steps to reproduce

1. Set a cookie with a string containing spaces.
2. Set a hidden input's default value to this cookie.
3. View this field's submitted value in the submissions tab, emails, etc. It will remain encoded with `%20` replacing spaces.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

4.5.5

### Plugin version

dev-craft-4

### Multi-site?

No

### Additional context

I tried fixing this with the `Hidden::EVENT_MODIFY_DEFAULT_VALUE` event, like so:

```
Event::on(Hidden::class, Hidden::EVENT_MODIFY_DEFAULT_VALUE, function(ModifyFieldValueEvent $event)
{
    $event->value = rawurldecode($event->value);
});
```

But that doesn't seem to change anything.

**Comment by engram-design** [comment]
I can't seem to replicate this, sorry. I've tested setting a cookie value from PHP and JS, and there's no encoding going on. Can you confirm how you're setting your cookie value? I can see the values of the field storing content without special characters.

Or, do you mean the value you're setting in the cookie is already url-encoded, and you want to decode it?

**Comment by amphibian** [comment]
I'm using the nystudio107 Cookies plugin to set the cookie in a template.

`{% do setCookie('formBookTitle', book.title) %}`

The raw value I'm setting is _not_ URL-encoded - it's just an entry title.

The resulting stored cookie value in my browser is obviously URL-encoded as PHP does that by default, but it's either not being decoded when being fetched (which I don't actually think is possible since I think PHP decodes it automatically on fetch?), or being re-encoded somewhere?

Example result in my Formie email notification:

> **Book**
> I%20Will%20Live%20for%20Both%20of%20Us

**Comment by engram-design** [comment]
Right! So there's a difference in behaviour when using the Cookies plugin. I was testing using `$_COOKIE` to set and that's no issue. From my understanding, when accessing a cookie via `$_COOKIE` PHP is supposed to decode it automatically like you say.

But I think I see the culprit. We use JS to fetch the value, which returns the encoded value. The plugin also uses `setcookie()` to see the cookie, which encodes values using `urlencode()`. Looks like the recommended approach is to use `setrawcookie()`. In any case, we can deal with it much better no matter what.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.36"`.

**Comment by engram-design** [comment]
Fixed in [2.0.37](https://github.com/verbb/formie/releases/tag/2.0.37)

**Comment by amphibian** [comment]
Perfect, thanks Josh!
